kubernetes/lib/kubernetes/configuration.rb

```diff
@@ -153,6 +153,35 @@ def self.default
       @@default ||= Configuration.new
     end
 
+    def self.default_config()
+      # KUBECONFIG environment variable
+      result = Configuration.new()
+      kc = "#{ENV['KUBECONFIG']}"
+      if File.exist?(kc)
+        k_config = KubeConfig.new(kc)
+        k_config.configure(result)
+        return result
+      end
+      # default home location
+      kc = "#{ENV['HOME']}/.kube/config"
+      if File.exist?(kc)
+        k_config = KubeConfig.new(kc)
+        k_config.configure(result)
+        return result
+      end
+      # In cluster config
+      if InClusterConfig::is_in_cluster()
```

Comment:
same here `InClusterConfig.in_cluster?`

Also you can omit the `()` if there are no params.

Maybe we should add a linter to this too. I will add a ticket.

Reply:
done.